### PR TITLE
Linting and various cleaning

### PR DESCRIPTION
**Pull request by CBroz1**
### Submission checklist

- [X] Reviewed all `Files changed` prior to selecting `Create Pull Request`.
- [ ] NA: Ran and passed all pytests, and ran the pipeline for features not covered by tests.
- [X] Ran pre-commit checks, including black, flake8, etc.
- [ ] NA: Updated the changelog and if making a new release, updated the version.py.
- [ ] NA: Updated documentation, including concepts.md, tutorials.md, and relevant docstrings.
- [ ] NA:After merge, I will push a new tag to upstream (test* prefix if no new release).

This PR does not add any new features and will not make any differences to the site, so several of the above have been skipped. This PR is submitted to standardize this repo and make my forthcoming feature PR easier to review

### Submission details

In this pull request, I make the following changes:
- Standardize file maintenance via a markdown linter and pre-commit checks
   - gitignore: add vscode folders
   - markdownlint.yaml: specify rules to ignore
   - pre-commit-config.yaml: pre-commit checks including import sorter, black and flake8
   - cspell.json - spell check extension for vscode. add author names as permitted words
   - mkdocs.yaml - add footnote feature
- Observe markdown linting. Particularly adding linebreaks after headings and limiting line length
  - changelog, contributing, readme, citation.md, index.md - line breaks and line length
  - mkdocs.yaml - spell checking
- Observe black/flake8
  - miniscope.py - sort imports, remove extra docstring spaces, intent to multiple of 4 spaces
  - version.py - single to double quotes 
  - setup.py - sort imports, add 'ignore F821' for undefined variable
- adjust changelog to be symbolic link

### CODE REVIEW

File: element_miniscope/miniscope.py [hunk 1]
```diff
@@ -689,9 +688,9 @@ class Block(dj.Part):
             block_y (longblob): y_start and y_end of this block in pixels.
             block_x (longblob): x_start and x_end of this block in pixels.
             y_shifts (longblob): y motion correction shifts for every frame in pixels.
-            x_shifts (longblob): x motion correction shifta for every frame in pixels. 
+            x_shifts (longblob): x motion correction shifta for every frame in pixels.
```
Comment (kushalbakshi) [inline element_miniscope/miniscope.py]:
```suggestion
            x_shifts (longblob): x motion correction shifts for every frame in pixels.
```

**Review by kushalbakshi — DISMISSED** [review]
I just found one small typo in the docstrings but the bulk of the content in this PR looks good. Once committed, we're ready to merge it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
 # Contribution Guidelines
 
-This project follows the [DataJoint Contribution Guidelines](https://docs.datajoint.io/python/community/02-Contribute.html). Please reference the link for more full details.
\ No newline at end of file
+This project follows the [DataJoint Contribution
+Guidelines](https://docs.datajoint.io/python/community/02-Contribute.html). Please
+reference the link for more full details.
```
Comment (kabilar) [inline CONTRIBUTING.md]:
```suggestion
This project follows the [DataJoint Contribution
Guidelines](https://datajoint.com/docs/community/contribute/). Please
reference the link for more full details.
```

**Review by kabilar — CHANGES_REQUESTED** [review]
Thanks @CBroz1!  I have provided a suggestion above.

**Review by kabilar — APPROVED** [review]

**Pull request merged**